projects/network-simulations-with-shadow.md

```diff
@@ -0,0 +1,87 @@
+
+Issues to investigate include, but are not limited to:
+- [PeerDAS][3]: Figure out if the current specs are resilient enough.
+- Future DAS approaches: Research on resilient DHTs is ongoing, and simulations could help with their evaluation.
+- [Lighthouse's message queues][6]: Lighthouse has some long-standing issues with memory consumption on nodes with low bandwidth, as messages do not get dequeued fast enough if they are in the queue for too long.
+- [Reth is working on optimizations on discv5][7], which need to be benchmarked.
+
+### Phase 3: Cleanup
+
+If the experiments prove to be useful, I want to extend the simulation framework to be as convenient to use as possible and support simulations with all major clients, so that the client teams can easily run their own simulations.
+
+This phase will start towards the end of Cohort 5, or even afterwards. Its length will depend on the progress on `ethereum-shadow` during phases 1 and 2, but I expect at least two weeks.
+
+## Possible challenges
+
+It is hard to estimate how much effort will be required to get Shadow to work with client software. It might be necessary to develop extensions to Shadow itself as more system calls might need to be implemented to properly run Reth and Lighthouse.
+
+Depending on the experiment, it might be necessary to simulate with a large amount of nodes. For economic reasons, such experiments must be carefully thought out and prepared. Running simulations on that scale might also cause performance and stability issues that need to be fixed.
+
+It remains to be seen whether the Shadow-based approach provides significant advantages over the Kurtosis-based testing approach mentioned above. Advantages might include reduced resource requirements (as Kubernetes is not involved) and the possibility to run simulations faster than real-time (or slower for large simulations). Disadvantages might include lower realism with Shadow (which uses its own TCP stack and compromises on realism for determinism).
+
+## Goal of the project
+
+- Figure out if Shadow is a good tool to test Ethereum networking
+- Prepare a nice framework for sophisticated simulations
+- Improve client networking and help with the specs for the upcoming hardfork
+- Empower all the client teams and researchers to use Shadow if it proves to be useful
+
+## Collaborators
+
+### Fellows
+
+- [Daniel Knopik](https://github.com/dknopik)
+
+No collaboration with other fellows is planned at this time, however, experiments involving the projects of other interested fellows will be strongly considered.
```

Comment:
I think @AshliaYan might be a good person to collaborate, afaik she has network security training

Reply:
Yes, I have worked with the Shadow framework before. I might also use it for simulating the rated list DHT network and analyzing network security threats. We can have more discussions on that if you are interested.